//*****************************************************************************
// Copyright 2023 Intel Corporation
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//*****************************************************************************

syntax = "proto2";
package mediapipe;

import "mediapipe/framework/calculator.proto";

message ModelAPIInferenceCalculatorOptions {
  extend mediapipe.CalculatorOptions {
    // https://github.com/google/mediapipe/issues/634 have to be unique in app
    // no rule to obtain this
    optional ModelAPIInferenceCalculatorOptions ext = 113473743;
    }
    map<string, string> tag_to_input_tensor_names = 1;
    map<string, string> tag_to_output_tensor_names = 2;
    // repeated gives optional option by default
    repeated string input_order_list = 3;
    repeated string output_order_list = 4;
}
